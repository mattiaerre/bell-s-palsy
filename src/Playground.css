.Number {
  background-color: transparent;
  border: 0;
  border-bottom: 2px solid darkgray;
  border-radius: 0;
  font-size: 4rem;
  font-family: 'Lucida Sans', 'Lucida Sans Regular', 'Lucida Grande',
    'Lucida Sans Unicode', Geneva, Verdana, sans-serif;
  text-align: center;
  width: 3rem;
}

@media (max-width: 900px) {
  .Number {
    font-size: 3rem;
    width: 2rem;
  }
}

@media (max-width: 400px) {
  .Number {
    font-size: 2rem;
    width: 1rem;
  }
}

.Playground {
  display: -webkit-flex;
  display: flex;
  -webkit-justify-content: space-around;
  justify-content: space-around;
  margin: 3rem auto 0 auto;
  max-width: 300px;
}
