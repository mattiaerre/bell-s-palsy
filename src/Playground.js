import { useEffect, useState } from 'react';
import makeChecker from './makeChecker';
import './Playground.css';

function Playground({ callback, password }) {
  const checker = makeChecker(password);

  const [valid, setValid] = useState(false);

  useEffect(() => {
    if (valid) {
      callback('authorized');
    }
  }, [callback, valid]);

  return (
    <section className="Playground">
      {['b', 'e', 'l', 'l'].map((placeholder, index) => (
        <input
          className="Number"
          onChange={({ target: { value } }) => {
            checker.add(index, value);
            setValid(checker.valid());
          }}
          placeholder={placeholder}
          key={index}
          maxLength="1"
          type="text"
        />
      ))}
    </section>
  );
}

export default Playground;
