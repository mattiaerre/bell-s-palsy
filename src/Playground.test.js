import { render } from '@testing-library/react';
import Playground from './Playground';

test('snapshot', () => {
  const { asFragment } = render(<Playground callback={jest.fn()} />);
  expect(asFragment()).toMatchSnapshot();
});

test('authorized', () => {
  // TODO
});
