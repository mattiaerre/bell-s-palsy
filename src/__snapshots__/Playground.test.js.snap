// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <section
    class="Playground"
  >
    <input
      class="Symbol"
      maxlength="1"
      placeholder="b"
      type="text"
    />
    <input
      class="Symbol"
      maxlength="1"
      placeholder="e"
      type="text"
    />
    <input
      class="Symbol"
      maxlength="1"
      placeholder="l"
      type="text"
    />
    <input
      class="Symbol"
      maxlength="1"
      placeholder="l"
      type="text"
    />
  </section>
</DocumentFragment>
`;
